src/test/java/com/fasterxml/jackson/databind/deser/UnknownPropertiesTest.java

```diff
@@ -0,0 +1,65 @@
+package com.fasterxml.jackson.databind.deser;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.StreamReadConstraints;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+
+public class UnknownPropertiesTest extends BaseMapTest
```

Comment:
I take it this for testing performance aspects? That is, does not test correctness but how long run takes?

If so, I wonder if it rather belongs under `src/test/java/perf/`?

Unless we can make it run reliably against timeout or something.
I was also wondering about if mock library (we seem to use PowerMock) would allow checking for calls to `BigInteger` constructor or something (there should be none).

Reply:
* It does show that NumberInput.parseBigInteger does not get called (and it did before this PR change)
* this can only be proved using debug mode and setting a breakpoint in NumberInput.parseBigInteger
* the test runs a lot faster than before (400ms as opposed to 4.5s on my laptop)
* unfortunately, testing something based on how fast it runs is going to cause flakiness (people with slow computers, CI builds where you have a build agent that is suffering from contention with other builds/build agents, etc.)
* the NumberInput code is not injectable - Jackson does not use dependency injection or an equivalent mechanism - so it is not possible to inject a mocked out NumberInput implementation

Reply:
@pjfanning Right, NumberInput is not injectable. But I have been amazed by what mocking frameworks can do. I may be wrong but I think it is actually possible to mock static methods too, f.ex:

https://www.baeldung.com/mockito-mock-static-methods

Maybe I can take some time to try to figure this one out as there are cases where it'd be super useful to actually verify call sequences like here.